libfreerdp/codec/nsc.c

```diff
@@ -303,34 +303,39 @@ NSC_CONTEXT* nsc_context_new(void)
 	/* init optimized methods */
 	NSC_INIT_SIMD(context);
 	return context;
-error_PlanePool:
-	free(context->priv);
-error_priv:
-	free(context);
+error:
+	nsc_context_free(context);
 	return NULL;
 }
 
 void nsc_context_free(NSC_CONTEXT* context)
 {
-	int i;
+	size_t i;
 
-	for (i = 0; i < 4; i++)
+	if (!context)
+		return;
+
+	if (context->priv)
 	{
-		if (context->priv->PlaneBuffers[i])
+		for (i = 0; i < 4; i++)
 		{
-			free(context->priv->PlaneBuffers[i]);
-			context->priv->PlaneBuffers[i] = NULL;
+			if (context->priv->PlaneBuffers[i])
```

Comment:
`free(NULL)` is ok according to the specification of `free` and I think there's no need for `context->priv->PlaneBuffers[i] = NULL;` as we're freeing the whole context after that (except of course if you're afraid of nasty code that would have aliased the context, but that's self-protecting code then)

Reply:
@hardening that code has not changed, only added the two `if` before

Reply:
@akallabeth yes sure that's not your code.

Reply:
@hardening adjusted the code accordingly (and fixed `profiler_free` with `NULL` arguments too)